main.cpp

```diff
@@ -19,13 +19,26 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : _rewriter(rewriter) {};
+    void run(const MatchFinder::MatchResult& Result) override {

```

Comment:
- How to prevent parentheses from being added if they are already present in the (int)(f) code?
- How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?
- How to prevent changing the C style cast code in macros?

Reply:
> * How to prevent parentheses from being added if they are already present in the (int)(f) code?

Использовать проверку на `isa<ParenExpr>`:
```c++
        if (!isa<ParenExpr>(cast_expression->getSubExprAsWritten()->IgnoreImpCasts())) {
            new_text_begin.append(")");
            _rewriter.InsertTextAfterToken(cast_expression->getEndLoc(), ")");
        }
```


> * How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?

Использовать проверку на `CK_ToVoid`:
```cpp
        if (cast_expression->getCastKind() == CK_ToVoid) return;
```


> * How to prevent changing the C style cast code in macros?

Использовать проверку на 
```cpp
if (cast_expression->getExprLoc().isMacroID()) return;
```

Включил эти изменения в свой код